pkg/util/log/gophercloud.go

```diff
@@ -0,0 +1,70 @@
+package log
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+)
+
+func NewLoggingProviderClient(endpoint string, logger kitlog.Logger) (*gophercloud.ProviderClient, error) {
+	providerClient, err := openstack.NewClient(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	transport := providerClient.HTTPClient.Transport
+	if transport == nil {
+		transport = &http.Transport{}
```

Comment:
Shouldn't we do `http.DefaultTransport` here?

Reply:
good catch